### ISSUE TITLE
reload: reloading shakes out runahead tasks

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-flow#5825]
There's a nasty interaction between reload and runahead.

```ini
[scheduler]                    
    allow implicit tasks = True    
                               
[scheduling]                   
    initial cycle point = 2000
    runahead limit = P5D       
    [[graph]]                  
        P1D = """              
            a => b             
        """                    
                               
[runtime]                      
    [[root]]                   
        script = sleep 500 
```

When this workflow starts, the tasks 2000 through 2005 are submitted:

```
2023-11-17T10:05:39Z INFO - [2000/a submitted job:01 flows:1] => running
2023-11-17T10:05:39Z INFO - [2000/a running job:01 flows:1] health: execution timeout=None, polling intervals=PT1H,...
2023-11-17T10:05:39Z INFO - [2002/a submitted job:01 flows:1] => running
2023-11-17T10:05:39Z INFO - [2002/a running job:01 flows:1] health: execution timeout=None, polling intervals=PT1H,...
2023-11-17T10:05:39Z INFO - [2003/a submitted job:01 flows:1] => running
2023-11-17T10:05:39Z INFO - [2003/a running job:01 flows:1] health: execution timeout=None, polling intervals=PT1H,...
2023-11-17T10:05:39Z INFO - [2004/a submitted job:01 flows:1] => running
2023-11-17T10:05:39Z INFO - [2004/a running job:01 flows:1] health: execution timeout=None, polling intervals=PT1H,...
2023-11-17T10:05:39Z INFO - [2005/a submitted job:01 flows:1] => running
2023-11-17T10:05:39Z INFO - [2005/a running job:01 flows:1] health: execution timeout=None, polling intervals=PT1H,...
```

But if I reload it, the task 2006 submits:

```
2023-11-17T10:05:40Z INFO - Reloading task definitions.
2023-11-17T10:05:40Z INFO - [2000/a running job:01 flows:1] reloaded task definition
2023-11-17T10:05:40Z WARNING - [2000/a running job:01 flows:1] active with pre-reload settings
2023-11-17T10:05:40Z INFO - [2001/a running job:01 flows:1] reloaded task definition
2023-11-17T10:05:40Z WARNING - [2001/a running job:01 flows:1] active with pre-reload settings
2023-11-17T10:05:40Z INFO - [2002/a running job:01 flows:1] reloaded task definition
2023-11-17T10:05:40Z WARNING - [2002/a running job:01 flows:1] active with pre-reload settings
2023-11-17T10:05:40Z INFO - [2003/a running job:01 flows:1] reloaded task definition
2023-11-17T10:05:40Z WARNING - [2003/a running job:01 flows:1] active with pre-reload settings
2023-11-17T10:05:40Z INFO - [2004/a running job:01 flows:1] reloaded task definition
2023-11-17T10:05:40Z WARNING - [2004/a running job:01 flows:1] active with pre-reload settings
2023-11-17T10:05:40Z INFO - [2005/a running job:01 flows:1] reloaded task definition
2023-11-17T10:05:40Z WARNING - [2005/a running job:01 flows:1] active with pre-reload settings
2023-11-17T10:05:40Z INFO - [2006/a waiting(runahead) job:00 flows:1] reloaded task definition
2023-11-17T10:05:40Z INFO - LOADING job data
2023-11-17T10:05:40Z INFO - [2006/a waiting(runahead) job:00 flows:1] => waiting
2023-11-17T10:05:40Z INFO - [2006/a waiting job:00 flows:1] => waiting(queued)
```

Reload again and 2007 submits:

```
2023-11-17T10:05:45Z INFO - Reloading task definitions.
...
2023-11-17T10:05:45Z INFO - [2007/a waiting(runahead) job:00 flows:1] => waiting
2023-11-17T10:05:45Z INFO - [2007/a waiting job:00 flows:1] => waiting(queued)
2023-11-17T10:05:45Z INFO - Reload completed.
```

Reload again and 2008 submits:

```
2023-11-17T10:05:48Z INFO - Reloading task definitions.
...
2023-11-17T10:05:49Z INFO - [2008/a waiting(runahead) job:00 flows:1] => waiting
2023-11-17T10:05:49Z INFO - [2008/a waiting job:00 flows:1] => waiting(queued)
```

And so on.

**Comment by oliver-sanders** [comment]
The issue seems to be caused by compute_runahead counting cycles with runahead tasks as active.

**Comment by hjoliver** [comment]
Closed by #5893
